packages/vda-name-client/src/blockchain/blockchainApi.ts

```diff
@@ -21,12 +23,11 @@ import { JsonRpcProvider } from '@ethersproject/providers';
  * @param web3Options: Web3 configuration depending on call type. Values from vda-did-resolver
  */
 export interface NameClientConfig {
-    identifier?: string;
-    signKey?: string;
-    chainNameOrId?: string | number;
-
-    callType: Web3CallType;
-    web3Options: Web3SelfTransactionConfig | Web3MetaTransactionConfig;
+    network: EnvironmentType
+    did?: string
+    signKey?: string
+    callType?: Web3CallType
+    web3Options?: Web3SelfTransactionConfig | Web3MetaTransactionConfig
```

Comment:
For the conditional type, it'd actually be a [discriminated union](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions).

```ts
export type NameClientConfig = {
  network: EnvironmentType
  did?: string
  signKey?: string
} & (
  | {
      callType?: "web3";
      web3Options?: Web3SelfTransactionConfig;
    }
  | {
      callType?: "gasless";
      web3Options?: Web3MetaTransactionConfig;
    }
);
```

Reply:
I get a syntax error with that, so have left for now.

Happy to accept a PR if you can get it to work.